h2o-algos/src/test/java/hex/tree/gbm/GBMInteractionConstraintsTest.java

```diff
@@ -0,0 +1,266 @@
+    }
+
+    @Test
+    public void testInteractionUnknownColError() {
+        try {
+            String[][] interactions = new String[2][];
+            interactions[0] = new String[]{"RACE"};
+            interactions[1] = new String[]{"42"};
+            testInteractionConstraintsSimple(interactions);
+        } catch (H2OModelBuilderIllegalArgumentException ex){
+            assert ex.getMessage().contains("ERRR on field: _interaction_constraints: Invalid interaction constraint - there is no column '42' in the training frame.");
+        }
+    }
+
+    @Test
+    public void testInteractionResponseColError() {
+        try {
+            String[][] interactions = new String[2][];
+            interactions[0] = new String[]{"RACE"};
+            interactions[1] = new String[]{"CAPSULE"};
+            testInteractionConstraintsSimple(interactions);
+        } catch (H2OModelBuilderIllegalArgumentException ex){
+            assert ex.getMessage().contains("ERRR on field: '_interaction_constraints': Column with the name 'CAPSULE' is used as response column and cannot be used in interaction.");
+        }
+    }
+
+    @Test
+    public void testInteractionIgnoredColError() {
+        try {
+            String[][] interactions = new String[2][];
+            interactions[0] = new String[]{"RACE"};
+            interactions[1] = new String[]{"ID"};
+            testInteractionConstraintsSimple(interactions);
+       } catch (H2OModelBuilderIllegalArgumentException ex){
+            assert ex.getMessage().contains("ERRR on field: _interaction_constraints: Column with the name 'ID' is set in ignored columns and cannot be used in interaction.");
+        }
+    }
+    
+    public void testInteractionConstraintsSimple(String[][] interactionConstraints) {
+        testInteractionConstraintsSimple(interactionConstraints, 3, 10);
+    }
+    
+    public void testInteractionConstraintsSimple(String[][] interactionConstraints, int maxDepth, int ntrees) {
+        Frame fr = null;
+        GBMModel model = null;
+        Scope.enter();
+        try {
+            final String response = "CAPSULE";
+            final String testFile = "smalldata/logreg/prostate.csv";
+            fr = parseTestFile(testFile)
+                    .toCategoricalCol(response);
+            Scope.track(fr);
+            DKV.put(fr);
+
+            GBMModel.GBMParameters parms = new GBMModel.GBMParameters();
+            parms._response_column = response;
+            parms._ignored_columns = new String[]{"ID"};
+            parms._train = fr._key;
+            parms._ntrees = ntrees;
+            parms._max_depth = maxDepth;
+            parms._seed = 1234L;
+            parms._interaction_constraints = interactionConstraints;
+
+            model = new GBM(parms).trainModel().get();
+
+            String[] names = Arrays.copyOfRange(fr.names(), 2, fr.names().length);
+            checkTrees(model, names);
+        } finally {
+            if (model != null) model.delete();
```

Comment:
Can we use Scope.track_generic(model); and get rid of this line? Its working this way in other tests